src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,322 @@
+        {
+            _mainLayout = new Box(parent);
+            return _mainLayout.Handle;
+        }
+
+        void Initialize()
+        {
+            AlignmentX = -1;
+            AlignmentY = -1;
+            WeightX = 1;
+            WeightY = 1;
+
+            _mainLayout.RepeatEvents = true;
+
+            _drawer = new Box(this);
+            _drawer.Show();
+
+            _contentBox = new Box(this);
+            _contentBox.PropagateEvents = false;
+            _contentBox.Show();
+
+            _iconBox = new Box(this);
+            //TODO remove it
+            _iconBox.BackgroundColor = EColor.Pink;
+            _iconBox.Show();
+
+            _drawer.PackEnd(_contentBox);
+            _drawer.PackEnd(_iconBox);
+
+            _mainLayoutGesture = new GestureLayer(_mainLayout);
+            _mainLayoutGesture.Attach(_mainLayout);
+
+            _mainLayoutGesture.SetMomentumCallback(GestureLayer.GestureState.Start, OnContentDragStarted);
+            _mainLayoutGesture.SetMomentumCallback(GestureLayer.GestureState.End, OnContentDragEnded);
+            _mainLayoutGesture.SetMomentumCallback(GestureLayer.GestureState.Abort, OnContentDragEnded);
+
+            _drawerGesture = new GestureLayer(_drawer);
+            _drawerGesture.Attach(_drawer);
+
+            _drawerGesture.SetMomentumCallback(GestureLayer.GestureState.Move, OnDrawerDragged);
+            _drawerGesture.SetMomentumCallback(GestureLayer.GestureState.End, OnDrawerDragEnded);
+            _drawerGesture.SetMomentumCallback(GestureLayer.GestureState.Abort, OnDrawerDragEnded);
+
+            _mainLayout.SetLayoutCallback(OnLayout);
+            _drawer.SetLayoutCallback(OnContentLayout);
+
+            RotaryEventManager.Rotated += OnRotateEventReceived;
+        }
+
+        async void OnRotateEventReceived(EventArgs args)
+        {
+            if (_fadeInCancelTokenSource != null)
+            {
+                _fadeInCancelTokenSource.Cancel();
+            }
+
+            _fadeInCancelTokenSource = new CancellationTokenSource();
+            var token = _fadeInCancelTokenSource.Token;
+
+            if (!_isOpen)
+            {
+                await HideAsync(_drawer);
+
+                _ = ShowAsync(_drawer, cancelltaionToken:token);
+            }
+        }
+
+        void OnLayout()
+        {
+            var geometry = Geometry;
+            geometry.Y = (_isOpen) ? 0 : (geometry.Height - _iconHeight);
+
+            _mainLayout.Geometry = Geometry;
+            _drawer.Geometry = geometry;
+        }
+
+        void OnContentLayout()
+        {
+            var geometry = _drawer.Geometry;
+            geometry.Height = _iconHeight;
+
+            _contentBox.Geometry = _drawer.Geometry;
+            _iconBox.Geometry = geometry;
+
+            _iconBox.StackAbove(_contentBox);
+        }
+
+        void OnContentDragStarted(GestureLayer.MomentumData moment)
+        {
+            if (_fadeInCancelTokenSource != null)
```

Comment:
_fadeInCancelTokenSource?.Cancel();
```suggestion
_fadeInCancelTokenSource?.Cancel();
_fadeInCancelTokenSource = null;

```
_fadeInCancelTokenSource = null;

Reply:
Fixed.